### PR TITLE
ftp: truncate first segment if over max length - v3

### PR DESCRIPTION
**Pull request by jasonish**
The first segment was not limited to the configured maximum line length
allowing it to be up to 65k. This could result in the next input length
being negative, which while handled properly by the code, did trigger a
debug validation assertion.

The fix is to be consistent and apply the limit to the first segment as
well, which does ensure the input_len could never be less than 0.

Ticket https://redmine.openinfosecfoundation.org/issues/5281

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on tlpw1_files_sha256.
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk | | |
| .flow.memuse | 509338688 | 539921728 | 94.34% | 

Pipeline 7170

File: src/app-layer-ftp.c [hunk 1]
```diff
@@ -397,13 +400,18 @@ static int FTPGetLineForDirection(FtpState *state, FtpLineState *line_state)
          * if we see fragmentation then it's definitely something you
          * should alert about */
         if (line_state->current_line_db == 0) {
-            line_state->db = FTPMalloc(state->input_len);
+            int32_t input_len = state->input_len;
```
Comment (catenacyber) [inline src/app-layer-ftp.c:403]: Why is `input_len` signed ?

Comment (jasonish) [reply src/app-layer-ftp.c:403]: Only because its source, `state->input_len` is in32_t.

Comment (catenacyber) [reply src/app-layer-ftp.c:403]: Indeed, but why is `state->input_len` signed in the first place ?

Comment (jasonish) [reply src/app-layer-ftp.c:403]: > Indeed, but why is `state->input_len` signed in the first place ?

```
 ¯\_(ツ)_/¯
```

Comment (jasonish) [reply src/app-layer-ftp.c:403]: > So, I take this is not the good opportunity fo fix this... Should I create a ticket for it for later ?

Yeah, I'd rather keep the fix as focused as possible for now.

Comment (catenacyber) [reply src/app-layer-ftp.c:403]: https://redmine.openinfosecfoundation.org/issues/5311

**Comment by victorjulien** [comment]
Merged in #7349, thanks!

**Pull request closed without merging**